### PR TITLE
Fix panic when getting a long container list before building the container

### LINKED ISSUE #1391: panic when getting a long container list before building the container
### Steps to reproduce

```
[janderson@rocky main]$ sudo wwctl container import docker://alpine:latest
[janderson@rocky main]$ sudo wwctl container list --long
```

### Error message

```
ERROR  : stat /var/lib/warewulf/provision/container/alpine:latest.img: no such file or directory
ERROR  : stat /var/lib/warewulf/provision/container/alpine:latest.img.gz: no such file or directory
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0xd9fc44]

goroutine 1 [running]:
github.com/warewulf/warewulf/internal/pkg/api/container.ContainerList()
	/builddir/build/BUILD/warewulf-4.5.1test1+188.g373bfcbd.dirty/internal/pkg/api/container/container.go:359 +0x7b4
github.com/warewulf/warewulf/internal/app/wwctl/container/list.GetCommand.CobraRunE.func1(0x4000288900?, {0x1134307?, 0x4?, 0x1134227?})
	/builddir/build/BUILD/warewulf-4.5.1test1+188.g373bfcbd.dirty/internal/app/wwctl/container/list/main.go:21 +0xa8
github.com/spf13/cobra.(*Command).execute(0x400047e300, {0x4000618bf0, 0x1, 0x1})
	/builddir/build/BUILD/warewulf-4.5.1test1+188.g373bfcbd.dirty/vendor/github.com/spf13/cobra/command.go:983 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x1abdf20)
	/builddir/build/BUILD/warewulf-4.5.1test1+188.g373bfcbd.dirty/vendor/github.com/spf13/cobra/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(0x40000021a0?)
	/builddir/build/BUILD/warewulf-4.5.1test1+188.g373bfcbd.dirty/vendor/github.com/spf13/cobra/command.go:1039 +0x1c
main.main()
	/builddir/build/BUILD/warewulf-4.5.1test1+188.g373bfcbd.dirty/cmd/wwctl/main.go:13 +0x24
```

### Information on your system

```
[janderson@rocky main]$ wwctl version
wwctl version:	 4.5.1test1+188.g373bfcbd.dirty-1
rpc version: apiPrefix:"rc1"  apiVersion:"1"  warewulfVersion:"4.5.1test1+188.g373bfcbd.dirty-1"
```

```
[janderson@rocky main]$ cat /etc/os-release 
NAME="Rocky Linux"
VERSION="9.4 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.4"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.4 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.4"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.4"
```

### General information

- [X] I have run `wwctl version` and reported the contents of `/etc/os-release`
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have captured and reported relevant error messages and logs

### PR DESCRIPTION
**Pull request by mslacken**
close #1391

### CODE REVIEW

File: internal/pkg/testenv/testenv.go [hunk 1]
```diff
@@ -131,6 +137,10 @@ func (env *TestEnv) WriteFile(t *testing.T, fileName string, content string) {
 	defer f.Close()
 	_, err = f.WriteString(content)
 	assert.NoError(t, err)
+	err = os.Chtimes(env.GetPath(fileName),
+		time.Date(2006, time.February, 1, 3, 4, 5, 0, time.UTC),
+		time.Date(2006, time.February, 1, 3, 4, 5, 0, time.UTC))
```
Comment (anderbubble) [inline internal/pkg/testenv/testenv.go:142]: What is the significance of this date?

File: internal/app/wwctl/container/list/main_test.go [hunk 2]
```diff
@@ -83,13 +71,12 @@ WW_INTERNAL: 0
...
 			stdout := <-stdoutC
-			assert.NotEmpty(t, stdout, "os.stdout should not be empty")
-			if !strings.Contains(stdout, tt.stdout) {
-				t.Errorf("Got wrong output, got:\n '%s'\n, but want:\n '%s'\n", stdout, tt.stdout)
-				t.FailNow()
-			}
+			assert.Equal(t, tt.output, stdout)
+			assert.Equal(t,
+				strings.ReplaceAll(strings.TrimSpace(tt.output), " ", ""),
+				strings.ReplaceAll(strings.TrimSpace(stdout), " ", ""))
```
Comment (anderbubble) [inline internal/app/wwctl/container/list/main_test.go:78]: I don't really like this pattern of testing output with all whitespace removed. But I don't know that I'd stand in the way of a pr for it at this stage: we need the CLI to be better about whitespace.

File: internal/app/wwctl/container/list/main.go [hunk 3]
```diff
@@ -28,6 +28,10 @@ func CobraRunE(vars *variables) func(cmd *cobra.Command, args []string) (err err
 				for i := 0; i < len(containerInfo); i++ {
 					createTime := time.Unix(int64(containerInfo[i].CreateDate), 0)
 					modTime := time.Unix(int64(containerInfo[i].ModDate), 0)
+					if vars.utc {
+						createTime = time.Unix(int64(containerInfo[i].CreateDate), 0).UTC()
+						modTime = time.Unix(int64(containerInfo[i].ModDate), 0).UTC()
```
[the remark this answers is not in the document]
Comment (mslacken) [reply internal/app/wwctl/container/list/main.go]: Right the new option is introduced for testing as didn't find an easy way to add UTC via environment variables,so this was faster.

Yes,
```
createTime = createTime.UTC()
modTime = modTime.UTC()
```
works as well, changed that.

File: internal/pkg/testenv/testenv.go [hunk 4]
```diff
@@ -116,6 +117,11 @@ func (env *TestEnv) GetPath(fileName string) string {
 func (env *TestEnv) MkdirAll(t *testing.T, dirName string) {
 	err := os.MkdirAll(env.GetPath(dirName), 0755)
 	assert.NoError(t, err)
+	/*
+		cmd := exec.Command("touch", "-t", "200601030405", env.GetPath(dirName))
+		err = cmd.Run()
+		assert.NoError(t, err)
+	*/
```
[the remark this answers is not in the document]
Comment (mslacken) [reply internal/pkg/testenv/testenv.go:124]: Done

File: internal/pkg/testenv/testenv.go (hunk 1 above)
Comment (mslacken) [reply internal/pkg/testenv/testenv.go:142]: Nothing, or it was the first example on stack overflow.  And its 1.2.3.4.5.6

File: internal/app/wwctl/container/list/main_test.go [hunk 5]
```diff
@@ -2,79 +2,67 @@ package list
...
+	// need to touch the files, so that the creation date of directories is constant
+	cmd := exec.Command("touch", "-d", "2000-01-02 03:04:05 UTC",
+		env.GetPath(path.Join(testenv.WWChrootdir, "test/rootfs")),
+		env.GetPath(path.Join(testenv.WWChrootdir, "test")))
+	fmt.Printf("path: %s\n", env.GetPath(path.Join(testenv.WWChrootdir, "test/rootfs/bin/sh")))
+	err := cmd.Run()
 	assert.NoError(t, err)
+	defer env.RemoveAll(t)
 	warewulfd.SetNoDaemon()
+	// wwlog.SetLogLevel(wwlog.DEBUG)
```
[the remark this answers is not in the document]
Comment (mslacken) [reply internal/app/wwctl/container/list/main_test.go]: Done

File: internal/app/wwctl/container/list/main_test.go (hunk 2 above)
Comment (mslacken) [reply internal/app/wwctl/container/list/main_test.go:78]: I am trying to make the output equal with whitespaces, but this seems a bit more robust.

**Comment by anderbubble** [comment]
Rebased on main.

File: internal/app/wwctl/container/list/main.go [hunk 6]
```diff
@@ -28,6 +28,10 @@ func CobraRunE(vars *variables) func(cmd *cobra.Command, args []string) (err err
 				for i := 0; i < len(containerInfo); i++ {
 					createTime := time.Unix(int64(containerInfo[i].CreateDate), 0)
 					modTime := time.Unix(int64(containerInfo[i].ModDate), 0)
+					if vars.utc {
+						createTime = createTime.UTC()
+						modTime = modTime.UTC()
+					}
```
Comment (anderbubble) [inline internal/app/wwctl/container/list/main.go:34]: I realized we don't actually need to add this; we can just set `TZ=UTC` when running the test suite.

I'll try to add a commit that does that for your comment.

**Review by anderbubble — APPROVED** [review]
@mslacken I move the UTC selection from a new arg to the standard environment variable, but I kept it in a separate commit in case you object. Let me know what you think.

**Comment by mslacken** [comment]
For me TZ variable didn't work so I went down the way with the command line args. So for you the TZ works?

**Pull request merged**